src/internal_queue.rs

```diff
@@ -1 +1,42 @@
+struct SimpleQueue<T> {
+    payload: Vec<T>,
+    pos: usize,
+}
 
+impl<T> SimpleQueue<T>
+where
+    T: Copy,
```

Comment:
Why we want `T` to be `Copy`? It seems to me that passing `T` itself instead of reference `&T` in `push` is more natural.

Reply:
Because it was what the original code did.  But yes, it's so unnatural and complicated to write in Rust, so I'll fix it.

Reply:
I think removing `Copy` is better in the normal Rust libraries, but considering this is a internal code and is only used in the implementation of the maxflow algorithm (only for integer and pair of integers), I also think we can restrict the level of abstraction. Original C++ implementation are not using move semantics, either. For more general purposes we should use `VecDeque` in my opinion. At least it doesn't have enough interface to use with non-`Copy` type, e.g. no way to get the ownership of the queued value back.
As an alternative, we can use `T` directly instead of using references `&T` if `T: Copy` to make it more natural.

Reply:
Sorry for late response....

Reply:
I see.  At least I changed `&T` to `T` in `push` method.  Also it seemed that removing `T: Copy` does not do any harm, so I simply removed it for now.